src/sentry/ingest/transaction_clusterer/datasource/redis.py

```diff
@@ -121,3 +116,31 @@ def _should_store_transaction_name(event_data: Mapping[str, Any]) -> bool:
         return False
 
     return True
+
+
+def _bump_rule_lifetime(project: Project, event_data: Mapping[str, Any]) -> None:
+    from sentry.ingest.transaction_clusterer import rules as clusterer_rules
+
+    applied_rules = event_data.get("_meta", {}).get("transaction", {}).get("", {}).get("rem", {})
+    if not applied_rules:
+        return
+
+    # XXX(iker): we're accessing project options here, and it represents a high
+    # (read) load.  However, since we're cleaning the redis store often, it's
+    # the only way to fetch the rules.
+    # Not checking if new rules exist means other strings could be added to the
+    # rules storage, which can explode the size of a project config.
+    stored_rules = clusterer_rules._get_rules(project)
+
+    for applied_rule in applied_rules:
+        # There are two types of rules:
+        # Transaction clustering rules  -- ["<pattern>", "<action>"]
+        # Other rules                   -- ["<reason>", "<action>", <from>, <to>]
+        # We are only looking at the transaction clustering rules, so checking
+        # for the length of the array should be enough.
+        if len(applied_rule) == 2:
+            pattern = applied_rule[0]
+            if pattern in stored_rules:
```

Comment:
As an alternative approach here I thought about checking if the pattern looks like a glob, but I find this approach more solid and easier to deal with.

Reply:
This also prevents "rogue" patterns from landing in project options (through merge). Imagine we allow users to delete rules through the UI at some point. Then we do not want to re-introduce the rule just because Relay still applies them. The project option should always remain the single source of truth.

Reply:
> Imagine we allow users to delete rules through the UI at some point.

In this case, we'd introduce another check to ensure the pattern is not in that list. I'm not sure if this is the point you were trying to do with your comment -- I'm not sure if I understood it.

Reply:
> In this case, we'd introduce another check to ensure the pattern is not in that list

I think `pattern in stored_rules` is exactly that check. `stored_rules` will never be populated by rules other than those discovered by the clusterer, so we are good. 👍